modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2MultiAttributeUserFilterTestCase.java

```diff
@@ -0,0 +1,396 @@
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "User " +
+                "has not been created successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        String usernameFromResponse = ((JSONObject) responseObj).get(USER_NAME_ATTRIBUTE).toString();
+        assertEquals(usernameFromResponse, userName);
+
+        String userId;
+        userId = ((JSONObject) responseObj).get(ID_ATTRIBUTE).toString();
+        userIds.add(userId);
+        assertNotNull(userId);
+    }
+
+    @Test(dependsOnMethods = "testCreateUser")
+    public void testCreateGroup() throws Exception {
+
+        createGroup(USERNAMES.subList(0, 3), userIds.subList(0, 3), GROUPNAMES.get(0));
+        createGroup(USERNAMES.subList(2, USERNAMES.size()), userIds.subList(2, USERNAMES.size()), GROUPNAMES.get(1));
+    }
+
+    private void createGroup(List<String> userNames, List<String> userIDs, String groupName) throws IOException {
+
+        HttpPost request = new HttpPost(getGroupPath());
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        JSONObject rootObject = new JSONObject();
+
+        JSONArray members = new JSONArray();
+        for (int i = 0; i < userNames.size(); i++) {
+            JSONObject member = new JSONObject();
+            member.put(SCIM2BaseTestCase.DISPLAY_ATTRIBUTE, userNames.get(i));
+            member.put(VALUE_PARAM, userIDs.get(i));
+            members.add(member);
+        }
+
+        rootObject.put(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE, groupName);
+
+        rootObject.put(SCIM2BaseTestCase.MEMBERS_ATTRIBUTE, members);
+
+        StringEntity entity = new StringEntity(rootObject.toString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "User " +
+                "has not been created successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        String groupNameFromResponse = ((JSONObject) responseObj).get(SCIM2BaseTestCase.DISPLAY_NAME_ATTRIBUTE)
+                .toString();
+        //TODO: groupsName comes from this is <Userstore_domain>/displayName
+        assertTrue(groupNameFromResponse.contains(groupName));
+
+        String groupId = ((JSONObject) responseObj).get(ID_ATTRIBUTE).toString();
+        groupIds.add(groupId);
+        assertNotNull(groupId);
+    }
+
+    @Test(dependsOnMethods = "testCreateGroup")
+    public void testMultiAttributeFilterUser() throws Exception {
+
+        //Validate username and claim filter
```

Comment:
use correct format for comments. Add space after // and fullstop at the end. 
Please check other places as well.

Reply:
fixed with https://github.com/wso2/product-is/pull/6381